File: catalog/backends/php-laravel/patterns/SlowQueryMysqlRepository.php
<?php

use Illuminate\Support\Facades\DB;

// Pattern: DBM slow query — artificial delay via MySQL SLEEP
// Adapt: replace 'your_table' with a domain entity table

function findAllSlowMysql(): array
{
    return DB::select("SELECT *, SLEEP(0.3) FROM your_table ORDER BY created_at DESC LIMIT 50");
}

File: catalog/backends/ruby-rails/patterns/slow_query_mysql_repository.rb
# Pattern: DBM slow query — artificial delay via MySQL SLEEP
# Adapt: replace 'your_table' with a domain entity table

def self.find_all_slow_mysql
  sql = "SELECT *, SLEEP(0.3) FROM your_table ORDER BY created_at DESC LIMIT 50"
  ActiveRecord::Base.connection.execute(sql).to_a
end
